### DISCUSSION TITLE
Using Redis with Mutiny - Composing asynchronous actions

### DISCUSSION THREAD

**Discussion by giscus** [quarkusio/quarkus#22542] [Quarkus Blog/Website]
# Using Redis with Mutiny - Composing asynchronous actions

Quarkus: Supersonic Subatomic Java

https://quarkus.io/blog/mutiny-redis/

**Comment by cdennison** [comment]
Clement - great guide but a few things are missing based on my experience in production with Redis and Muntiny:

1. You should never calls the Redis "keys" command in production see here - https://redis.io/commands/KEYS

Instead there's an iterative approach which is more complex in Mutiny https://stackoverflow.com/questions/69815030/is-recursion-necessary-in-quarkus-mutiny

2. Another challenge with Redis is that even when using the iterative approach above it's easy to exhaust the connection pool so I've found operations need to be executed sequentially using the "call" method e.g.

    List<String> keys = new ArrayList<>();
    for (Integer i = 0; i < 100; i++) {
      keys.add("KEY" + i.toString());
    }

    Multi.createFrom().iterable(keys)
        .onItem().call(item -> reactiveRedisClient.get(item))
        .subscribe().with(
            redisValue -> {
              LOGGER.info("Got Redis value: " + redisValue);
            });
